packages/data-lake-api/src/DataLakeDatabase.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017-2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { types as pgTypes } from 'pg';
+import knex from 'knex';
+
+import { getConnectionConfig } from './getConnectionConfig';
+
+// turn off parsing of timestamp (not timestamptz), so that it stays as a sort of "universal time"
+// string, independent of timezones, rather than being converted to local time
+pgTypes.setTypeParser(pgTypes.builtins.TIMESTAMP, val => val);
+
+export class DataLakeDatabase {
+  constructor() {
+    const connectToDatabase = async () => {
+      this.connection = await knex({
+        client: 'pg',
+        connection: getConnectionConfig(),
+      });
+      return true;
+    };
+    this.connectionPromise = connectToDatabase();
+  }
+
+  async closeConnections() {
+    return this.connection.destroy();
+  }
+
+  async waitUntilConnected() {
```

Comment:
Do we need this? This is useful for our primary db because we want to keep trying to connect to the db in case it blips. But for simple data fetching, the connection should either be there or show an error to the user I'd say.

Reply:
Hmmm yeah fair point! I'll scrap it 👍